app/controllers/concerns/shopify_app/ensure_has_session.rb

```diff
@@ -6,14 +6,20 @@ module EnsureHasSession
 
     included do
       include ShopifyApp::Localization
-      include ShopifyApp::LoginProtection
+
+      if ShopifyApp.configuration.use_new_embedded_auth_strategy?
+        include ShopifyApp::RetrieveSessionFromTokenExchange
+        around_action :activate_shopify_session
```

Comment:
I repeated it so that it's explicit in which action is called, if we leave this as a shared/common action outside of this condition, then we're implicitly forcing the method name to be the same between the 2 concerns. Which would also work, but I feel like less implicit dependency the better? 🤔

Reply:
Yeah I agree - less magic, more good.